src/integrationTest/java/uk/gov/hmcts/darts/common/repository/EventRepositoryTest.java

```diff
@@ -34,23 +33,46 @@ void testEventProcessing() {
 
         List<Integer> eventIdsToBeProcessed1 = eventRepository.getCurrentEventIdsToBeProcessed(Pageable.ofSize(1));
         Assertions.assertEquals(1, eventIdsToBeProcessed1.size());
-
-        Integer eventPkid = eventRepository.getTheLatestCreatedEventPrimaryKeyForTheEventId(eventIdsToBeProcessed1.get(0));
+        EventRepository.EventIdAndHearingIds eventPkid = eventRepository.getTheLatestCreatedEventPrimaryKeyForTheEventId(eventIdsToBeProcessed1.getFirst());
         eventRepository.updateAllEventIdEventsToNotCurrentWithTheExclusionOfTheCurrentEventPrimaryKey(
-            List.of(eventPkid), List.of(eventIdsToBeProcessed1.get(0)));
-        Assertions.assertTrue(eventStub.isOnlyOneOfTheEventIdSetToCurrent(eventIdMap.get(eventIdsToBeProcessed1.get(0))));
+            eventPkid.getEveId(), eventPkid.getEventId(), eventPkid.getHearingIds());
+        Assertions.assertTrue(eventStub.isOnlyOneOfTheEventIdSetToCurrent(eventIdMap.get(eventIdsToBeProcessed1.getFirst())));
 
         List<Integer> eventIdsToBeProcessed2 = eventRepository.getCurrentEventIdsToBeProcessed(Pageable.ofSize(1));
-        Integer eventPkidSecond = eventRepository.getTheLatestCreatedEventPrimaryKeyForTheEventId(eventIdsToBeProcessed2.get(0));
+        EventRepository.EventIdAndHearingIds eventPkidSecond =
+            eventRepository.getTheLatestCreatedEventPrimaryKeyForTheEventId(eventIdsToBeProcessed2.getFirst());
         eventRepository.updateAllEventIdEventsToNotCurrentWithTheExclusionOfTheCurrentEventPrimaryKey(
-            List.of(eventPkidSecond), List.of(eventIdsToBeProcessed2.get(0)));
+            eventPkidSecond.getEveId(), eventPkidSecond.getEventId(), eventPkidSecond.getHearingIds());
 
         Assertions.assertEquals(1, eventIdsToBeProcessed1.size());
-        Assertions.assertTrue(eventIdMap.containsKey(eventIdsToBeProcessed2.get(0)));
+        Assertions.assertTrue(eventIdMap.containsKey(eventIdsToBeProcessed2.getFirst()));
         Assertions.assertNotEquals(eventIdsToBeProcessed1, eventIdsToBeProcessed2);
 
         // ENSURE WE DONT PROCESS THE THIRD BATCH I.E. THE ZERO EVENT ID
         List<Integer> eventIdsToBeProcessed3 = eventRepository.getCurrentEventIdsToBeProcessed(Pageable.ofSize(1));
         Assertions.assertTrue(eventIdsToBeProcessed3.isEmpty());
     }
+
+    @Test
+    void testEventProcessingWithMultipleHearings() {
```

Comment:
Any chance you can make it a bit clearer what this test is actually trying to achieve? It's not clear to me what this is trying to do.
This example [PR](https://github.com/mario-paniccia/darts-api-fork/pull/1/files) gives examples of the sort of format we want to use (although it's not been implemented throughout the project, it would be good if we can try and align to this as much as possible).

Reply:
Updated :)